pkg/storage/buckets/prefixed.go

```diff
@@ -19,50 +18,50 @@ type prefixedObjStore struct {
 	prefix string
 }
 
-func (o *prefixedObjStore) Meta(ctx context.Context, p storj.Path) (meta objects.Meta, err error) {
+func (o *prefixedObjStore) Meta(ctx context.Context, path storj.Path) (meta objects.Meta, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	if len(p) == 0 {
+	if len(path) == 0 {
 		return objects.Meta{}, objects.NoPathError.New("")
 	}
 
-	m, err := o.o.Meta(ctx, path.Join(o.prefix, p))
+	m, err := o.o.Meta(ctx, storj.JoinPaths(o.prefix, path))
```

Comment:
`prefix + path` here as well

Reply:
The `prefix` in `prefixedObjStore` is always initialized with bucket name without a trailing slash. So we need `storj.JoinPaths` to add a slash between the bucket name and the actual object path.

Once we change how we store buckets in the PointerDB (i.e. the bucket name is not part of the object's path) the `prefixedObjStore` will become obsolete.